docs/porting/target/static_pinmap.md

```diff
@@ -0,0 +1,180 @@
+<h1 id="static-pinmap-port">Static Pinmap extension</h1>
+
+The **Static Pinmap extension** allows you to statically specify the peripheral configuration (pin, peripheral or function) in the HAL API function.
```

Comment:
Is this a HAL API, or is an extension something different? If it is an API, we should link to the Doxygen in the implementing section.

Reply:
The HAL API will be extended: `xxx_init_direct()` functions will be added (e.g. `spi_init_direct()`). Also, the driver's layer will be extended and we will have additional constructors that will use `xxx_direct()` HAL API.

Which link should be added?
https://os.mbed.com/docs/mbed-os/v5.14/apis/spi.html 

I'm not sure how these docs are updated. At the moment there is no constructor that uses `xxx_direct()` HAL API on the list. Will it be added automatically when static pinmap is merged into the master?

Reply:
Which HAL API will be extended? SPI?

When new APIs are added with the right Doxygen tags and doxyfile options (I think here: https://github.com/ARMmbed/mbed-os/blob/master/doxygen_options.json), they show up in the Doxygen: https://os.mbed.com/docs/mbed-os/v5.14/mbed-os-api-doxy/annotated.html

Reply:
The following HAL and Drivers (classes) APIs will be extended:
- SPI
- I2C
- Serial
- PWM
- AnalogIn
- AnalogOut
- CAN
- QSPI